source/components/ResolutionSlider.js

```diff
@@ -1,43 +1,111 @@
 import React from 'react'
-import {scaleLinear} from 'd3-scale'
-
-import {tileEdgeRange} from '../constants'
-
-const normalizeValue = scaleLinear()
-  .domain([tileEdgeRange.min, tileEdgeRange.max])
-  .range([0, 100])
+import {scaleLog} from 'd3-scale'
 
 export default class ResolutionSlider extends React.Component {
   constructor(props) {
     super(props)
-
     this.state = {
-      value: normalizeValue(tileEdgeRange.default),
+      value: 50,
+      typedValue: '',
+    }
+    this.normalizeValue = scaleLog().domain(this.props.metricDomain).range([1, 99])
+    this._triggerChangeFromText = this._triggerChangeFromText.bind(this)
+  }
+
+  componentDidMount() {
+    this._triggerChange(50, {normalized: true}) // reset slider to 50
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (JSON.stringify(nextProps) !== JSON.stringify(this.props)) {
```

Comment:
perhaps check the specific props that you are interesting in observing a change instead of props as a whole?